cpp/include/cudf/detail/scatter.cuh

```diff
@@ -425,14 +425,33 @@ std::unique_ptr<table> scatter(
                  });
 
   // We still need to call `gather_bitmask` even when the source columns are not nullable,
-  // as if the target has null_mask, that null_mask needs to be updated after scattering
+  // as if the target has null_mask, that null_mask needs to be updated after scattering.
   auto const nullable =
     std::any_of(source.begin(), source.end(), [](auto const& col) { return col.nullable(); }) or
     std::any_of(target.begin(), target.end(), [](auto const& col) { return col.nullable(); });
   if (nullable) {
     auto const gather_map = scatter_to_gather(
       updated_scatter_map_begin, updated_scatter_map_end, target.num_rows(), stream);
     gather_bitmask(source, gather_map.begin(), result, gather_bitmask_op::PASSTHROUGH, stream, mr);
+
```

Comment:
Naive question.
Where is the cuda stream sync between the `gather_bitmask` kernel launch and the usage of `result` on the host? I ask because I am thinking `null_count` might be modified by `gather_bitmask`

Reply:
In the line above, `source` is the (`const`) input while `result` will be modified inside `gather_bitmask`, with stream sync. That function `gather_bitmask` is kind of modifying-the-reference-argument, instead of returning a value.